### ISSUE TITLE
Documentation improvements for filters

### ISSUE THREAD

**Issue by ljahn** [lambdaloop/anipose#42]
**Is your feature request related to a problem? Please describe.**
I'd like to use filtering (2D as well as 3D with distance constraints) but the description of the filter parameters is insufficient. While it is clear how to enable filters and what general kind of operation they will perform, I cannot find accurate descriptions of the filter parameters.

**Describe the solution you'd like**
Please clarify these points in the documentation:
* What happens when a point is considered erroneous? Is this a requirement for a point to be corrected by a filter or do filters apply to all points in general?
* How do `scale_smooth` an `scale_length` work exactly? 
  * What kind of smoothing is applied and what exactly does `scale_smooth` mean? Is it smoothing kernel width, shape, ...?
  * How does `scale_length` work? I assume, if some constrained joint suddenly moves far away from its (by constraint) expected position, this somehow moves it back. How does `scale_length` specify this behaviour? Is it a maximum percentage, the constraint can get longer before action is taken, or is it the strength with which a constraint "pulls" on a joint, ...?
* Looking through the source code I find other parameters like `scale_length_weak`, `n_deriv_smooth`, `reproj_error_threshold` which also contribute to the optimization but are not mentioned at all in the docs. Please clarify what they mean.

**Additional context**
Maybe it's just me not finding these descriptions but I have looked [here](https://anipose.readthedocs.io/en/latest/params.html) (parameter list) and [here](https://anipose.readthedocs.io/en/latest/tutorial.html) (tutorial). As well as in the source code [here](https://github.com/lambdaloop/anipose/blob/master/anipose/triangulate.py) (anipose) and [here](https://github.com/lambdaloop/aniposelib/blob/master/aniposelib/cameras.py) (optimization function in aniposelib) which also gives no comments / explanation.

**Comment by lambdaloop** [comment]
You're right, these are not appropriately described. I'll have to put together an example for how each parameter affects the optimization in the docs

**Comment by Sam-Kelly** [comment]
Hi,
I'm also trying to use your 3D triangulation optimisation methods. As I understand it, `scale_smooth` and `scale_length` are the scale factors for the 'limb length' and 'temporal' loss described in the paper.

What were the parameters you used to get good quality 3D points? (Like in figure 7 in the paper). And how did you decide on those parameters?

**Comment by RiSpa5** [comment]
Hi, did you manage to understand more on the effects of different parameters for 2D and 3D filtering? As far as I can see there is still no documentation on the subject, or at least I wasn't able to find any!
